### PR TITLE
chore(jsonrpc): move `nearcore` type conversions to server

### PR DESCRIPTION
**Pull request by miraclx**
Tracking issue: https://github.com/near/nearcore/issues/6850

Move all conversions between `near-primitives` and `near-jsonrpc-primitives` to the server side. Freeing up a bunch of dependencies on `near-jsonrpc-primitives`.

### CODE REVIEW

**Comment by matklad** [comment]
I think this can be handled in a simpler way, without adding our own conversion trait/macro: 

```diff
diff --git a/chain/jsonrpc/src/errors.rs b/chain/jsonrpc/src/errors.rs
index f71d3217b..9082dc5cd 100644
--- a/chain/jsonrpc/src/errors.rs
+++ b/chain/jsonrpc/src/errors.rs
@@ -11,6 +11,10 @@ macro_rules! _rpc_try {
 
 pub(crate) use _rpc_try as rpc_try;
 
+pub(crate) struct InternalServerError {
+    pub(crate) message: String,
+}
+
 pub trait RpcFrom<T> {
     fn rpc_from(_: T) -> Self;
 }
@@ -36,9 +40,9 @@ impl RpcFrom<actix::MailboxError> for ServerError {
     }
 }
 
-impl RpcFrom<actix::MailboxError> for near_jsonrpc_primitives::types::blocks::RpcBlockError {
-    fn rpc_from(error: actix::MailboxError) -> Self {
-        Self::InternalError { error_message: error.to_string() }
+impl From<InternalServerError> for near_jsonrpc_primitives::types::blocks::RpcBlockError {
+    fn from(error: InternalServerError) -> Self {
+        Self::InternalError { error_message: error.message }
     }
 }
 
diff --git a/chain/jsonrpc/src/lib.rs b/chain/jsonrpc/src/lib.rs
index e76d9cbbc..7dac68f5d 100644
--- a/chain/jsonrpc/src/lib.rs
+++ b/chain/jsonrpc/src/lib.rs
@@ -5,6 +5,7 @@ use std::time::{Duration, Instant};
 use actix::Addr;
 use actix_cors::Cors;
 use actix_web::{get, http, middleware, web, App, Error as HttpError, HttpResponse, HttpServer};
+use errors::InternalServerError;
 use futures::Future;
 use futures::FutureExt;
 use prometheus;
@@ -984,12 +985,7 @@ impl JsonRpcHandler {
         near_jsonrpc_primitives::types::blocks::RpcBlockResponse,
         near_jsonrpc_primitives::types::blocks::RpcBlockError,
     > {
-        let block_view = rpc_try! {
-            self
-                .view_client_addr
-                .send(GetBlock(request_data.block_reference))
-                .await
-        }?;
+        let block_view = self.message_view_client(GetBlock(request_data.block_reference)).await??;
         Ok(near_jsonrpc_primitives::types::blocks::RpcBlockResponse { block_view })
     }
 
@@ -1188,6 +1184,18 @@ impl JsonRpcHandler {
         }?;
         Ok(validators)
     }
+
+    async fn message_view_client<M>(&self, msg: M) -> Result<M::Result, InternalServerError>
+    where
+        M: actix::Message + Send + 'static,
+        M::Result: Send,
+        ViewClientActor: actix::Handler<M>,
+    {
+        self.view_client_addr
+            .send(msg)
+            .await
+            .map_err(|it| InternalServerError { message: it.to_string() })
+    }
 }
 
 #[cfg(feature = "sandbox")]
```

There's a bit of logical copy-paste with handling actix messages. So we can extract *that* bit and do conversion once, rather than doing it on every call-site.

File: chain/jsonrpc/src/lib.rs [hunk 1]
```diff
@@ -664,29 +689,32 @@ impl JsonRpcHandler {
...
             loop {
-                let tx_status_result = self
-                    .view_client_addr
-                    .send(TxStatus {
+                let tx_status_result = message_actor(&self.view_client_addr, TxStatus {
                         tx_hash,
                         signer_account_id: account_id.clone(),
                         fetch_receipt,
                     })
                     .await;
```
Comment (miraclx) [inline chain/jsonrpc/src/lib.rs:697]: I wonder if `client_send` is a better name. 🤔 

```suggestion
                let tx_status_result = client_send(&self.view_client_addr, TxStatus {
                        tx_hash,
                        signer_account_id: account_id.clone(),
                        fetch_receipt,
                    })
                    .await;
```

Comment (matklad) [reply chain/jsonrpc/src/lib.rs:697]: I'd probably even introduce `.client_send` and `view_client_send`

File: chain/jsonrpc-primitives/Cargo.toml [hunk 2]
```diff
@@ -13,22 +13,16 @@ description = "This crate hosts structures for the NEAR JSON RPC Requests, Respo
...
-near-chain-configs = { path = "../../core/chain-configs" }
-near-client-primitives = { path = "../client-primitives" }
 near-crypto = { path = "../../core/crypto" }
-near-metrics = { path = "../../core/metrics" }
 near-primitives = { path = "../../core/primitives" }
-near-network-primitives = { path = "../../chain/network-primitives" }
-near-primitives-core = { path = "../../core/primitives-core" }
+near-chain-configs = { path = "../../core/chain-configs" }
 near-rpc-error-macro = { path = "../../tools/rpctypegen/macro" }
+near-client-primitives = { path = "../client-primitives" }
```
Comment (frol) [inline chain/jsonrpc-primitives/Cargo.toml:25]: On the initial read, I thought we introduced new cruel dependencies `chain-configs` and `client-primitives`, but in fact, they just got shuffled a bit

**Review by frol — APPROVED** [review]

File: chain/jsonrpc-primitives/Cargo.toml (hunk 2 above)
Comment (miraclx) [reply chain/jsonrpc-primitives/Cargo.toml:25]: Few more PRs and they'll be gone for good.

**Review by matklad — APPROVED** [review]

**Pull request merged**